src/headerUtils.js

```diff
@@ -153,29 +154,31 @@ module.exports = function(env) {
       throw new Error('mapData: Invalid batchSize.');
     }
 
-    var ix = env.coroutine.mapDataFetch ?
+    // var ix = env.coroutine.mapDataFetch ?
+    //     env.coroutine.mapDataFetch(data, batchSize, a) :
+    //     null;
+    var _data = env.coroutine.mapDataFetch ?
         env.coroutine.mapDataFetch(data, batchSize, a) :
-        null;
+        data;
```

Comment:
Following on from our chat yesterday, here's some code that demonstrates the property we need to preserve:

```js
// The code that follows requires this definition adding to e.g.
// headerUtils.js:

// function getAddress(s, k, a) {
//   return k(s, a);
// }

var model = function() {
  display('start');
  mapData({data: 'abc'.split(''), batchSize: 1}, function(x) {
    display('address: ' + getAddress() + ' | x: ' + x);
  });
};

// When running the model, the address within the observation function
// corresponds to the current value of x.
model();

// start
// address: _159_158_$$0_156 | x: a
// address: _159_158_$$1_156 | x: b
// address: _159_158_$$2_156 | x: c

// When using mini batches with elbo, the observation function will
// only be called on a subset of the data. When the observation
// function is called with some particular value of x, the address should
// be the same as the address used for this x when not doing mini batches,
// as above. Like so:

Optimize({model, steps: 5});

// start
// address: _160_158_$$1_156 | x: b
// start
// address: _160_158_$$2_156 | x: c
// start
// address: _160_158_$$1_156 | x: b
// start
// address: _160_158_$$0_156 | x: a
// start
// address: _160_158_$$2_156 | x: c
```
If you see a way to do it that doesn't involve returning both the indices and data from `mapDataFetch` that's fine with me!